src/cli/commands/test/cache.rs

```diff
@@ -21,6 +31,52 @@ pub mod cache {
 		pub hash: String,
 	}
 
+	fn is_valid_cairo_contract(contract_path: &PathBuf) -> Result<(), CacheError> {
+		// check if contract_path have .cairo extension
+		let extension = contract_path
+			.extension()
+			.ok_or(CacheError::InvalidContractExtension(format!(" ")))?;
+		// assert extension to be cairo
+		if extension != "cairo" {
+			// convert osStr to string
+			return Err(CacheError::InvalidContractExtension(
+				extension.to_str().to_owned().unwrap().to_string(),
+			))
+		}
+		Ok(())
+	}
+
+	pub fn get_cache_path(
+		contract_path: &PathBuf,
+		root_dir: &PathBuf,
+	) -> Result<PathBuf, CacheError> {
+		// check if contract_path have .cairo extension
+		is_valid_cairo_contract(contract_path)?;
+		let cache_dir = dirs::cache_dir().ok_or(CacheError::CacheDirNotSupportedError)?;
```

Comment:
could you make the cache_dir error into a transparent error?

Reply:
i'm still struggling, since `cache_dir` returns as `Option` 

and in the existing code it was handled this way `https://github.com/open-dust/cairo-foundry/blob/main/src/compile.rs#L79`

Reply:
Fair point! good to me